### DISCUSSION TITLE
Add the ability for resources (In my case plugins) to be paid

### DISCUSSION THREAD

**Discussion by ToberoCat** [modrinth/.github#251] [Feature requests]
Hi, first of all, thanks for creating this amazing resource sharing platform. I'm really glad about the current monetization system for free resources.
 
### The feature request

But sadly, I wasn't able to find the option to make a resource paid only, like I'm able to do with spigot plugins. I'd like to have the ability to not just upload resources for free, but also for a small price (Like 5$), as this increase the amount of money earned by developing and maintaining a resource.

For example a very time consuming plugin when it comes to maintenance targeted for a small player base will never generate as much revenue with just using the current monetization system.

### How it might be implemented

I think it should either be implemented by making a paid resources, where every the user once buys the resource and then has the lasting right to download the latest version.

Another way would be to make a specific version for an resource paid. Thereby allowing the resource to gain a attention by providing a free version, with only a limited feature set. The users that want the addional features should then be able to buy the premium version, which would result in a similar concept as above - Once a resource has been bought, the user will have the lasting right to download it as often as needed.

### Thoughts behind it

As I wasn't able to find such a premium resource system on modrinth by searching the wiki and github discussions, I now had to put up my premium plugin on spigot and not on modrinth, which I would by much prefer. I created a plugin that covers a niche, thereby it wouldn't attract masses, resulting in barely no ad-revenue.
I personally would love to see this feature on modrinth, ideally with something like the demo-like approach, where there's a free version with a very limited set of features.  

Thanks anyways

**Comment by AnOpenSauceDev** [comment]
To quote the [Minecraft EULA](https://www.minecraft.net/en-us/eula):

"Any Mods you create for Minecraft: Java Edition from scratch belong to you (including pre-run Mods and in-memory Mods) and you can do whatever you want with them, **as long as you don't sell them for money / try to make money from them** and so long as you don’t distribute Modded Versions of the game."

Charging money for a mod or offering paid features in a mod is directly prohibited by the EULA, so Modrinth will probably never distribute them.

**Reply by Gaming32** [reply]
I will note that a fair number of plugins don't use Minecraft at all, but rather communicate purely through the plugin server's API. Plugins like this could be considered not really mods, but that is walking a pretty fine line.

**Reply by AnOpenSauceDev** [reply]
The API just points to Minecraft's code. I'm pretty sure Mojang would get angry if you sold a datapack, which operates off similar principles.
